### PR TITLE
Cutscene and QoL fixes Pt. 2

### PR DESCRIPTION
**Pull request by MrMiguel211**
## Skippable Cutscenes
- Heavily improved Hive Totem, Energy Core and Ruined Courtyard cutscene skips.
  - The water now always ends up where it should be if you watched the whole thing.
- Improved Hive Totem intro cutscene skip timings.
- Fixed Observatory projector activation cutscene skip activating the projector twice if skipped the cutscene late.
- Fixed Flying Pirates moving up away from the player upon cutscene skip in Control Tower.
- Fixed Combat Visor being forced upon grabbing the pickup in Research Core for Competitive Cutscene mode.
- Added a "Doors Unlocked" HUD Memo in Control Tower once the fight is done for Competitive Cutscene mode.

## QoL
- Increased the size of the Load Trigger to Furnace Access in Energy Core.
- Adjusted the Hive Totem scan position to match it's PAL counterpart.
- Adjusted Ruined Courtyard water going down timer to better fit visuals.
- Fixed fog in Research Entrance disappearing after clearing the 1st Pass Pirates and before the 2nd Pass Shadow Pirates.
- Fixed Energy Core Puzzle music not playing again if left the room while the timer is going down.
- Fixed underwater sound playing even if you are not underwater in Energy Core.
  - The sound activation trigger now properly follows the water.
- Fixed Beetle music still playing if left Ruined Shrine through the bottom.
- Fixed sometimes getting stuck near the end of the Spider Track in Ruined Fountain.
- Fixed Save Station B playing it's music and interrupting Phendrana music if left the room too quickly.
- Removed overlooked music triggers in a few rooms.
- Observatory
  - Fixed incorrect music playing when the projector is active, it now plays it's designated music permanently upon room entry.
  - Fixed panel activation cutscene playing on 2nd Pass when the projector is already active.
  - Restored an unused particle effect for the projector.
  - The projector now loads activated instead of activating immediately upon load.

### CODE REVIEW

**Comment by toasterparty** [comment]
I'm going to use this PR to test the discord webhook. This is my first test comment

File: src/patches.rs
Comment (toasterparty) [inline src/patches.rs:1]: What are all these changes for? Is it purely cosmetic?

Comment (MrMiguel211) [reply src/patches.rs:1]: Correct, poison and water liquids looked really weird in-game, so I adjusted the properties to better match how most vanilla liquids look

File: generated/json_data/qol.jsonc [hunk 1]
```diff
@@ -3535,6 +3411,24 @@
...
+                        "524379":{
+                            "position": [235.663055, -51.395695, 32.689095]
+                        },
+                        "524380":{
+                            "position": [236.473007, -50.495583, 32.689095]
+                        },
+                        "524300":{
+                            "position": [236.726593, -48.492104, 32.689095]
+                        }
+                    },
```
Comment (toasterparty) [inline generated/json_data/qol.jsonc:3431]: This is a really good example of a change that pushes my comfort level in how much gameplay change I'm willing to accept for a QoL update and I had to think on it for a bit.

I'll accept this one, but be warry I won't accept any change which alters gameplay in a more significant manner than this.

**Review by toasterparty — APPROVED** [review]

File: generated/json_data/qol.jsonc (hunk 1 above)
Comment (duncathan) [reply generated/json_data/qol.jsonc:3431]: I think there needs to be a rule about making PRs more atomic tbh. many of these changes need to be considered separately, and wouldn't cause merge conflicts if split into separate PRs

Comment (toasterparty) [reply generated/json_data/qol.jsonc:3431]: The more stuff you put in a PR, the exponentially longer it will take me to approve and merge. No rule needed

Comment (MrMiguel211) [reply generated/json_data/qol.jsonc:3431]: > This is a really good example of a change that pushes my comfort level in how much gameplay change I'm willing to accept for a QoL update and I had to think on it for a bit.
> 
> I'll accept this one, but be warry I won't accept any change which alters gameplay in a more significant manner than this.

I mean you're the boss, If you're not comfortable about something then by all means I'll undo it, but I didn't think this was too much of a big change.

**Pull request merged**